File: src/lib/nodejs/fileUtils.js
/**
 * File utilities for handling project paths.
 */

const path = require('path');
const fs = require('fs');

/**
 * Get the project root directory.
 * @returns {string} Project root path
 */
function getProjectRoot() {
    return path.join(__dirname, '..', '..', '..');
}

/**
 * Get path to data file.
 * @param {string} subdir - Subdirectory under data/ (e.g., 'config', 'reference-data', 'raw-data')
 * @param {string} filename - Name of the file
 * @returns {string} Full path to the data file
 */
function getDataPath(subdir, filename) {
    return path.join(getProjectRoot(), 'data', subdir, filename);
}

/**
 * Get path to output file.
 * @param {string} subdir - Subdirectory under output/ (e.g., 'csv', 'images', 'logs', 'text')
 * @param {string} filename - Name of the file
 * @returns {string} Full path to the output file
 */
function getOutputPath(subdir, filename) {
    return path.join(getProjectRoot(), 'output', subdir, filename);
}

/**
 * Ensure directory exists.
 * @param {string} dirPath - Path to directory
 */
function ensureDir(dirPath) {
    if (!fs.existsSync(dirPath)) {
        fs.mkdirSync(dirPath, { recursive: true });
    }
}

module.exports = { getProjectRoot, getDataPath, getOutputPath, ensureDir };

File: src/lib/python/file_utils.py
"""File utilities for handling project paths."""

import os
from pathlib import Path


def get_project_root():
    """Get the project root directory."""
    return Path(__file__).parent.parent.parent.parent


def get_data_path(subdir, filename):
    """Get path to data file.

    Args:
        subdir: Subdirectory under data/ (e.g., 'config', 'reference-data', 'raw-data')
        filename: Name of the file

    Returns:
        Path object to the data file
    """
    return get_project_root() / "data" / subdir / filename


def get_output_path(subdir, filename):
    """Get path to output file.

    Args:
        subdir: Subdirectory under output/ (e.g., 'csv', 'images', 'logs', 'text')
        filename: Name of the file

    Returns:
        Path object to the output file
    """
    return get_project_root() / "output" / subdir / filename


def ensure_dir(path):
    """Ensure directory exists.

    Args:
        path: Path to directory (string or Path object)
    """
    os.makedirs(path, exist_ok=True)
